src/Microsoft.Diagnostics.NETCore.Client/DiagnosticsIpc/IpcEndpointConfig.cs

```diff
@@ -0,0 +1,126 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Runtime.InteropServices;
+
+namespace Microsoft.Diagnostics.NETCore.Client
+{
+    internal class IpcEndpointConfig
+    {
+        public enum PortType
+        {
+            Connect,
+            Listen
+        }
+
+        PortType _type;
+
+        public string Address { get; }
+
+        public bool IsConnectConfig {
+            get
+            {
+                if (!string.IsNullOrEmpty(Address) && _type == PortType.Connect)
+                    return true;
+                else
+                    return false;
+            }
+        }
+
+        public bool IsListenConfig {
+            get
+            {
+                if (!string.IsNullOrEmpty(Address) && _type == PortType.Listen)
+                    return true;
+                else
+                    return false;
+            }
+        }
+
+        const string NamedPipeSchema = "namedpipe";
+        const string UnixDomainSocketSchema = "uds";
+        const string NamedPipeDefaultIPCRoot = @"\\.\pipe\";
+        const string NamedPipeSchemaDefaultIPCRootPath = "/pipe/";
+
+        public IpcEndpointConfig(string address, PortType type)
+        {
+            try
+            {
+                if (Uri.TryCreate(address, UriKind.Absolute, out Uri parsedAddress))
+                {
+                    if (string.Equals(parsedAddress.Scheme, NamedPipeSchema, StringComparison.OrdinalIgnoreCase))
+                    {
+                        if (!RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+                            throw new ArgumentException($"{NamedPipeSchema} only supported on Windows.");
+
+                        address = parsedAddress.AbsolutePath;
+                    }
+                    else if (string.Equals(parsedAddress.Scheme, UnixDomainSocketSchema, StringComparison.OrdinalIgnoreCase))
+                    {
+                        if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+                            throw new ArgumentException($"{UnixDomainSocketSchema} not supported on Windows, use {NamedPipeSchema}.");
+
+                        address = parsedAddress.AbsolutePath;
+                    }
+                    else if (!string.IsNullOrEmpty(parsedAddress.Scheme))
+                    {
+                        throw new ArgumentException($"{parsedAddress.Scheme} not supported.");
+                    }
+                }
+            }
+            catch (InvalidOperationException)
```

Comment:
When parsing using UriKind.RelativeOrAbsolute it parse could succeed, but trying to access properties like Schema would throw InvalidOperationException if missing. Since then I switched this to UriKind.Absolute and then TryCreate is better to return failure if you don't have all needed components, so I believe we can drop this try/catch altogether.